### PR TITLE
Add PageBlob::Create

### PR DESCRIPTION
**Pull request by Jinming-Hu**

### CODE REVIEW

File: sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp [hunk 1]
```diff
@@ -259,6 +304,111 @@ class BlobRestClient
...
+            = Common::GetHeaderValue(response.getHeaders(), Common::k_HEADER_LAST_MODIFIED);
+        info.ContentMD5
+            = Common::GetHeaderValue(response.getHeaders(), Common::k_HEADER_CONTENT_MD5);
+        info.ContentCRC64
+            = Common::GetHeaderValue(response.getHeaders(), Common::k_HEADER_X_MS_CONTENT_CRC64);
+        info.SequenceNumber = std::stoull(Common::GetHeaderValue(
+            response.getHeaders(), k_HEADER_X_MS_BLOB_SEQUENCE_NUMBER));
+        info.ServerEncrypted
+            = Common::GetHeaderValue(response.getHeaders(), Common::k_HEADER_X_MS_SERVER_ENCRYPTED)
+            == "true";
```
Comment (katmsft) [inline sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp:362]: Should `true` also be defined as `k_HEADER_VALUE_TRUE` or `k_HEADER_TRUE`?

**Review by katmsft — APPROVED** [review]
LGTM

File: sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp [hunk 2]
```diff
@@ -7,27 +7,88 @@
...
+  P10,
+  P15,
+  P20,
+  P30,
+  P40,
+  P50,
+  P60,
+  P70,
+  P80,
+  Hot,
```
Comment (katmsft) [inline sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp:37]: I think page blob tier should be distinguished with block blob tier.

File: sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp [hunk 3]
```diff
@@ -83,7 +137,8 @@ class BlobRestClient
       {
         // TODO: ContainerUrl initialization.
         info.Etag = Common::GetHeaderValue(response.getHeaders(), Common::k_HEADER_ETAG);
-        info.LastModified = Common::GetHeaderValue(response.getHeaders(), Common::k_HEADER_LAST_MODIFIED);
+        info.LastModified
+            = Common::GetHeaderValue(response.getHeaders(), Common::k_HEADER_LAST_MODIFIED);
```
Comment (katmsft) [inline sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp:141]: >GetHeaderValue [](start = 22, length = 14)

Question: Is your maximum length set to 80? Do you think this is the value we should use in CLANG formatter?

File: sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp:37]: .Net Track2 doesn't distinguish them.

File: sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp (hunk 3 above)
Comment (Jinming-Hu) [reply sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp:141]: I  think it's 100

File: sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp:362]: okay

**Pull request merged**